<!-- -->
<launch>
  <arg name="gating_thres" default="9.22" />
  <arg name="gate_probability" default="0.99" />
  <arg name="detection_probability" default="0.9" />
  <arg name="distance_thres" default="99" />
  <arg name="life_time_thres" default="8" />
  <arg name="static_velocity_thres" default="3.0" />
  <arg name="tracker_input_topic" default="/detected_objects_range" />
  <arg name="tracker_output_jskbb_topic" default="/bounding_boxes_tracked" />
  <arg name="tracker_output_detected_topic" default="/tracker_output_detected_topic" />
  <arg name="is_benchmark" default="false" />

  <arg name="pointcloud_frame" default="/velodyne" />
  <arg name="tracking_frame" default="/world" />

  <!-- rosrun lidar_tracker euclidean_cluster _points_node:="" -->
  <node pkg="lidar_imm_ukf_pda_track" type="imm_ukf_pda" name="imm_ukf_pda" output="screen">
    <param name="gating_thres" value="$(arg gating_thres)" />
    <param name="gate_probability" value="$(arg gate_probability)" />
    <param name="detection_probability" value="$(arg detection_probability)" />
    <param name="distance_thres" value="$(arg distance_thres)" />
    <param name="life_time_thres" value="$(arg life_time_thres)" />
    <param name="static_velocity_thres" value="$(arg static_velocity_thres)" />
    <param name="pointcloud_frame" value="$(arg pointcloud_frame)" />
    <param name="tracking_frame" value="$(arg tracking_frame)" />
    <param name="is_benchmark" value="$(arg is_benchmark)" />

    <remap from="detected_objects_range" to="$(arg tracker_input_topic)" />
    <remap from="bounding_boxes_tracked" to="$(arg tracker_output_jskbb_topic)" />
    <remap from="detected_objects" to="$(arg tracker_output_detected_topic)" />
  </node>

  <node pkg="lidar_imm_ukf_pda_track" type="visualize_detected_objects" name="visualize_detected_objects" output="screen">
    <param name="pointcloud_frame" value="$(arg pointcloud_frame)" />

    <remap from="detected_objects" to="$(arg tracker_output_detected_topic)" />
  </node>
</launch>
